server/pulp/server/webservices/views/consumer_groups.py

```diff
@@ -0,0 +1,373 @@
+        units = body.get('units')
+        options = body.get('options')
+        task = factory.consumer_group_manager().install_content(consumer_group_id,
+                                                                units, options)
+        raise pulp_exceptions.OperationPostponed(task)
+
+    def update(self, request, consumer_group_id):
+        """
+        Update content (units) on the consumer in a consumer group.
+
+        Expected body: {units:[], options:<dict>}
+        where unit is: {type_id:<str>, unit_key={}} and the
+        options is a dict of update options.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_group_id: A consumer group ID.
+        :type consumer_group_id: str
+        :raises: OperationPostponed
+        """
+        body = request.body_as_json
+        units = body.get('units')
+        options = body.get('options')
+        task = factory.consumer_group_manager().update_content(consumer_group_id,
+                                                               units, options)
+        raise pulp_exceptions.OperationPostponed(task)
+
+    def uninstall(self, request, consumer_group_id):
+        """
+        Uninstall content (units) from the consumers in a consumer group.
+
+        Expected body: {units:[], options:<dict>}
+        where unit is: {type_id:<str>, unit_key={}} and the
+        options is a dict of uninstall options.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_group_id: A consumer group ID.
+        :type consumer_group_id: str
+        :raises: OperationPostponed
+        """
+        body = request.body_as_json
+        units = body.get('units')
+        options = body.get('options')
+        task = factory.consumer_group_manager().uninstall_content(consumer_group_id,
+                                                                  units, options)
+        raise pulp_exceptions.OperationPostponed(task)
+
+
+class ConsumerGroupBindingsView(View):
+    """
+    Views for repository binding to the group.
+    """
+    @auth_required(authorization.CREATE)
+    def post(self, request, consumer_group_id):
+        """
+        Create a bind association between the consumers belonging to the given
+        consumer group by id included in the URL path and a repo-distributor
+        specified in the POST body: {repo_id:<str>, distributor_id:<str>}.
+        Designed to be idempotent so only MissingResource is expected to
+        be raised by manager.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_group_id: The consumer group ID to bind.
+        :type consumer_group_id: str
+        :raises: MissingResource if group id does not exist
+        :raises: InvalidValue some parameters are invalid
+        :raises: OperationPostponed
```

Comment:
Can you say a little more about OperationPostponed? For intsance it is what is raised under normal operation. Making any similar fixes throughout the PR would be good too.